File: safety/snip/make_appender.cpp
// C++ make_appender

auto make_appender(std::vector<int> const& suffix) {
    return [&](std::vector<int>&& items) {
        return append(move(items), suffix);
    };
}

void test_make_appender() {
    std::vector<int> suffix{3, 4};
    auto append34 = make_appender(suffix);
    assert((std::vector<int>{1, 2, 3, 4} == append34({1, 2})));
}

File: safety/snip/make_appender.rs
// Rust make_appender

pub fn make_appender(suffix: &[i32]) -> impl Fn(Vec<i32>) -> Vec<i32> + '_ {
    |items| append(items, suffix)
}

fn test_make_appender() {
    let append34 = make_appender(&[3, 4]);
    assert_eq!(vec![1, 2, 3, 4], append34(vec![1, 2]));
}
